src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py

```diff
@@ -317,16 +325,21 @@ def _evaluate_point_validity_chart_area_flag_and_adjusted_speed_and_rate(
             # Todo: Need to QA this logic. Below minimum speed requires speed < minimum_speed, but this is not enforced?
             minimum_flow_rate_for_speed = self.minimum_rate_as_function_of_speed(speed_to_use)
             maximum_flow_rate_for_speed = self.maximum_rate_as_function_of_speed(speed_to_use)
-            if rate < minimum_flow_rate_for_speed:
+            # first update rate_to_use when below minimum flow if increase_rate_left_of_minimum_flow_assuming_asv
+            if rate_to_use < minimum_flow_rate_for_speed:
+                if increase_rate_left_of_minimum_flow_assuming_asv:
+                    rate_to_use = max(minimum_flow_rate_for_speed, rate_to_use)
```

Comment:
you already know that `rate_to_use < minimum_flow_rate_for_speed` ? Then just take `minimum_flow_rate_for_speed`?

Reply:
💡